src/RadhydroSimulation.hpp

```diff
@@ -1028,7 +1034,7 @@ auto RadhydroSimulation<problem_t>::advanceHydroAtLevel(amrex::MultiFab &state_o
 	auto dx = geom[lev].CellSizeArray();
 
 	// do Strang split source terms (first half-step)
-	addStrangSplitSourcesWithBuiltin(state_old_cc_tmp, lev, time, 0.5 * dt_lev);
+	auto burn_success_first = addStrangSplitSourcesWithBuiltin(state_old_cc_tmp, lev, time, 0.5 * dt_lev);
```

Comment:
I think I misunderstood your earlier comment. I think you should check if this is false here, and if so, return from this function early. There's no need to wait until the hydro update proper if the first Strang half-step already failed.